test/metabase/lib/filter_test.cljc

```diff
@@ -341,3 +343,22 @@
             query'       (lib/replace-clause query filter-clause external-op')]
         (is (=? {:stages [{:filters [[:!= {} [:field {} (meta/id :users :id)] 515]]}]}
                 query'))))))
+
+(deftest ^:parallel find-filter-for-legacy-filter-test
+  (let [query           (-> (lib/query meta/metadata-provider (meta/table-metadata :users))
+                            (lib/expression "expr" (lib/absolute-datetime "2020" :month)))
+        filterable-cols (lib/filterable-columns query)
+        [first-col]     filterable-cols
+        expr-col        (m/find-first #(= (:name %) "expr") (lib/filterable-columns query))
+        query           (->  query
+                             (lib/filter (lib/filter-clause
+                                          (first (lib/filterable-column-operators first-col))
+                                          first-col
+                                          515))
+                             (lib/filter (lib/filter-clause
+                                          (first (lib/filterable-column-operators expr-col))
+                                          expr-col
+                                          (meta/field-metadata :users :last-login))))]
+    (doseq [filter-clause (lib/filters query)]
+      (is (= filter-clause
+             (lib/find-filter-for-legacy-filter query (lib.convert/->legacy-MBQL filter-clause)))))))
```

Comment:
I'd be glad to see a test case for the other two cases: throwing for an ambiguous match and not finding anything.